src/pages/CustomerRequisitionPage.js

```diff
@@ -117,9 +137,94 @@ export const CustomerRequisition = ({
         sortKey={sortKey}
       />
     ),
-    [sortKey, isAscending]
+    [columns, sortKey, isAscending]
+  );
+
+  const ItemIndicatorToggles = useMemo(
+    () => [
+      {
+        text: programStrings.items,
+        isOn: !showIndicators,
+        onPress: onHideIndicators,
+      },
+      {
+        text: programStrings.indicators,
+        isOn: showIndicators,
+        onPress: onShowIndicators,
+      },
+    ],
+    [showIndicators]
+  );
+
+  const ItemIndicatorToggle = () => <ToggleBar toggles={ItemIndicatorToggles} />;
+
+  const ButtonSetSuppliedToRequested = useCallback(
+    () => (
+      <PageButton
+        style={globalStyles.topButton}
+        text={buttonStrings.use_requested_quantities}
+        onPress={onSetSuppliedToRequested}
+        isDisabled={isFinalised}
+      />
+    ),
+    [isFinalised]
+  );
+
+  const ButtonSetSuppliedToSuggested = useCallback(
+    () => (
+      <PageButton
+        style={globalStyles.topButton}
+        text={buttonStrings.use_suggested_quantities}
+        onPress={onSetSuppliedToSuggested}
+        isDisabled={isFinalised}
+      />
+    ),
+    [isFinalised]
+  );
+
+  const indicatorCodes = useMemo(() => indicators.map(indicator => indicator.code), [indicators]);
+  const selectedIndicatorCode = useMemo(() => selectedIndicator.code, [selectedIndicator]);
+
+  const IndicatorDropdown = useCallback(
+    () => (
+      <DropDown
+        values={indicatorCodes}
+        selectedValue={selectedIndicatorCode}
+        onValueChange={onSelectIndicator}
+      />
+    ),
+    [indicatorCodes, selectedIndicatorCode]
   );
 
+  const TopRightButtons = useCallback(() => {
+    const { horizontalContainer, verticalContainer } = globalStyles;
+    if (!usingIndicators) {
+      return (
+        <View style={horizontalContainer}>
+          <ButtonSetSuppliedToRequested />
+          <ButtonSetSuppliedToSuggested />
+        </View>
+      );
+    }
+    if (!showIndicators) {
+      return (
+        <View style={verticalContainer}>
+          <ItemIndicatorToggle />
+          <View style={horizontalContainer}>
+            <ButtonSetSuppliedToRequested />
+            <ButtonSetSuppliedToSuggested />
+          </View>
+        </View>
+      );
+    }
+    return (
+      <View style={verticalContainer}>
+        <ItemIndicatorToggle />
+        <IndicatorDropdown />
+      </View>
+    );
+  }, [usingIndicators, showIndicators]);
```

Comment:
Lots of repetition here.

was expecting something more like:

```
     <View style={verticalContainer}>
        {usingIndicators && <ItemIndicatorToggle />}
        <View style={horizontalContainer}>
          {!showIndicators ? (
            <>
              <ButtonSetSuppliedToRequested />
              <ButtonSetSuppliedToSuggested />
            </>
          ) : (
            <IndicatorDropdown />
          )}
        </View>
      </View>
```

Also the margin on the drop down is messing with the whole layout when you toggle

Reply:
This one is tricky to get a mix of concise but also readable. See e4184fe6fa057cce6933c7f45c705c4713ea8d51. Definitely more repetition than necessary, but I think it's a pretty good balance of readability and conciseness.